### ISSUE TITLE
Transient signal classes

### ISSUE THREAD

**Issue by jlaehne** [LumiSpy/lumispy#115]
We missed something when introducing the transient signal classes last year (and no one seems to have used them so far? Anyway there are no specific functions). At the moment, we have a set of Signal2D classes for transients:
```
LumiTransient | 2 | Luminescence | real | TRLumi, TR luminescence, time-resolved luminescence
CLTransient | 2 | TRCL | real | TR cathodoluminescence, time-resolved cathodoluminescence
PLTransient | 2 | TRPL | real | TR photoluminescence, time-resolved photoluminescence
```

That is correct for a streak camera image <x,y|E,t>, but not for a pure transient <x,y|t> (without spectral information).

The `...Transient` classes should in fact be Signal1D, while there should be an additional set of classes `...TransientSpectrum` for Streak camera images. Any oppinions? I would propose to fix it for the v0.2 release.

(see original discussion in https://github.com/LumiSpy/lumispy/issues/5#issuecomment-558125719 )

**Comment by jlaehne** [comment]
`LumiStreak` could also be an alternative or alias for `LumiTransientSpectrum` that is shorter. I find that the latter nicely shows that time-resolved and spectral information is combined, but it is kind of a long name for a signal_type compared with e.g. just `CL` and `Lumi`.

Any opinions @LMSC-NTappy @jordiferrero @gkusch?

**Comment by LMSC-NTappy** [comment]
>(and no one seems to have used them so far? Anyway there are no specific functions)

It's true. I myself mostly (only) use the general Signal1D/Signal2D stock classes. I guess when hyperspy 1.7 is out we'll add dedicated axes for spectroscopy setups (non-linear) and that's going to be a good start. The question is what kind of analysis is needed specifically for luminescence / TR spectra ? A "true luminescence color" widget would be nice...

>The ...Transient classes should in fact be Signal1D, while there should be an additional set of classes ...TransientSpectrum for Streak camera images. Any opinions? I would propose to fix it for the v0.2 release. 

Couldn't suggest anything better.  Btw I just submitted a pull request for loading hamamatsu streak map into hyperspy :)

>LumiStreak could also be an alternative or alias for LumiTransientSpectrum that is shorter.

Generally speaking I am in favour of keeping the name close to the data and not the detector, most transient spectra are recorded with TCSPCs I think...

**Comment by jordiferrero** [comment]
I personally have no strong opinions on the naming of the classes as I have never used them yet.
However, I am also in favour of using names clos to the data and not the detector.
